### PR TITLE
Add Entity Slicing using Entity Manifests to Cedar

### PR DESCRIPTION
**Pull request by oflatt**
## Description of changes

This PR allows users of Cedar to get an *entity slice* of an existing `Entities` store using the entity manifest.
Mainly this is useful for testing entity manifests. However, it may also be useful for users to load all entities in to memory, but only send a slice over the network to a server.

Later, I re-implemented most of this code in #1208. You can squash this code with all the PRs on top of it into one giant PR, or just review this and then review the replacement. 

PR stack:
https://github.com/cedar-policy/cedar/pull/1102 (Merged)
https://github.com/cedar-policy/cedar/pull/1105 (This PR)
https://github.com/cedar-policy/cedar/pull/1154
https://github.com/cedar-policy/cedar/pull/1156
https://github.com/cedar-policy/cedar/pull/1171
#1196 
https://github.com/cedar-policy/cedar/pull/1208

## Checklist for requesting a review

The change in this PR is (choose one, and delete the other options):

- [ ] A breaking change requiring a major version bump to `cedar-policy` (e.g., changes to the signature of an existing API).
- [ ] A backwards-compatible change requiring a minor version bump to `cedar-policy` (e.g., addition of a new API).
- [ ] A bug fix or other functionality change requiring a patch to `cedar-policy`.
- [ ] A change "invisible" to users (e.g., documentation, changes to "internal" crates like `cedar-policy-core`, `cedar-validator`, etc.)
- [x] A change (breaking or otherwise) that only impacts unreleased or experimental code.

I confirm that this PR (choose one, and delete the other options):

- [x] Updates the "Unreleased" section of the CHANGELOG with a description of my change (required for major/minor version bumps).
- [ ] Does not update the CHANGELOG because my change does not significantly impact released code.

I confirm that [`cedar-spec`](https://github.com/cedar-policy/cedar-spec) (choose one, and delete the other options):

- [ ] Does not require updates because my change does not impact the Cedar formal model or DRT infrastructure.
- [ ] Requires updates, and I have made / will make these updates myself. (Please include in your description a timeline or link to the relevant PR in `cedar-spec`, and how you have tested that your updates are correct.)
- [x] Requires updates, but I do not plan to make them in the near future. (Make sure that your changes are hidden behind a feature flag to mark them as experimental.)
- [ ] I'm not sure how my change impacts `cedar-spec`. (Post your PR anyways, and we'll discuss in the comments.)

### CODE REVIEW

**Comment by oflatt** [comment]
This PR stacks on top of #1102 
It looks like I can't select that PR as a base branch since it's from my fork of Cedar.

File: cedar-testing/src/cedar_test_impl.rs [hunk 1]
```diff
@@ -300,6 +304,34 @@ impl CedarTestImplementation for RustEngine {
...
+        let validator = Validator::new(schema.clone());
+        let validation_result = validator.validate(policies, ValidationMode::Strict);
+        if validation_result.validation_passed() {
+            let entity_manifest_res = compute_entity_manifest(schema, policies);
+
+            match entity_manifest_res {
+                Ok(entity_manifest) => {
+                    // PANIC SAFETY: slicing should succeed for tests
+                    #[allow(clippy::expect_used)]
+                    let entity_slice = entity_manifest
```
Comment (oflatt) [inline cedar-testing/src/cedar_test_impl.rs:318]: Are we allowed to panic in the testing infra here? We could return a new kind of TestResult or something like that.
Or perhaps I'm abusing the `is_authorized` function

**Comment by oflatt** [comment]
Looks like changing the API for cedar testing broke downstream dependencies build

File: cedar-testing/src/cedar_test_impl.rs (hunk 1 above)
Comment (oflatt) [reply cedar-testing/src/cedar_test_impl.rs]: Edit: Removed use of panic

File: cedar-policy-core/src/ast/entity.rs [hunk 2]
```diff
@@ -318,6 +318,49 @@ pub struct Entity {
 }
 
 impl Entity {
+    /// The implementation of [`Eq`] and [`PartialEq`] for
```
Comment (shaobo-he-aws) [inline cedar-policy-core/src/ast/entity.rs:321]: Can you add the feature flag to core as well?

**Review by shaobo-he-aws — CHANGES_REQUESTED** [review]

File: cedar-policy-core/src/ast/entity.rs (hunk 2 above)
Comment (shaobo-he-aws) [reply cedar-policy-core/src/ast/entity.rs:321]: Or generally, do you need changes to core at all? It seems you can put everything into the validator.

**Comment by shaobo-he-aws** [comment]
Closed in favor of a single PR.

**Pull request closed without merging**